src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs

```diff
@@ -86,61 +153,61 @@ public class WwwAuthenticateParameters
         /// <param name="scheme">Authentication scheme. Default is "Bearer".</param>
         /// <returns>The parameters requested by the web API.</returns>
         /// <remarks>Currently it only supports the Bearer scheme</remarks>
+        [System.ComponentModel.EditorBrowsable(EditorBrowsableState.Never)]  // hide confidential client on mobile
         public static WwwAuthenticateParameters CreateFromResponseHeaders(
             HttpResponseHeaders httpResponseHeaders,
             string scheme = "Bearer")
         {
             if (httpResponseHeaders.WwwAuthenticate.Any())
             {
-                // TODO: add POP support
-                AuthenticationHeaderValue bearer = httpResponseHeaders.WwwAuthenticate.First(v => string.Equals(v.Scheme, scheme, StringComparison.OrdinalIgnoreCase));
-                string wwwAuthenticateValue = bearer.Parameter;
-                return CreateFromWwwAuthenticateHeaderValue(wwwAuthenticateValue);
+                AuthenticationHeaderValue headerValue = httpResponseHeaders.WwwAuthenticate.FirstOrDefault(v => string.Equals(v.Scheme, scheme, StringComparison.OrdinalIgnoreCase));
+
+                if (headerValue != null)
+                {
+                    string wwwAuthenticateValue = headerValue.Parameter;
+                    var parameters = CreateFromWwwAuthenticateHeaderValue(wwwAuthenticateValue);
+                    parameters.AuthScheme = scheme;
+                    return parameters;
+                }
             }
 
-            return CreateWwwAuthenticateParameters(new Dictionary<string, string>());
+            return CreateWwwAuthenticateParameters(new Dictionary<string, string>(), string.Empty);
         }
 
         /// <summary>
         /// Creates parameters from the WWW-Authenticate string.
         /// </summary>
         /// <param name="wwwAuthenticateValue">String contained in a WWW-Authenticate header.</param>
         /// <returns>The parameters requested by the web API.</returns>
+        [System.ComponentModel.EditorBrowsable(EditorBrowsableState.Never)]
         public static WwwAuthenticateParameters CreateFromWwwAuthenticateHeaderValue(string wwwAuthenticateValue)
         {
-            if (string.IsNullOrWhiteSpace(wwwAuthenticateValue))
-            {
-                throw new ArgumentNullException(nameof(wwwAuthenticateValue));
-            }
-
-            IDictionary<string, string> parameters = CoreHelpers.SplitWithQuotes(wwwAuthenticateValue, ',')
-                .Select(v => ExtractKeyValuePair(v.Trim()))
-                .ToDictionary(pair => pair.Key, pair => pair.Value, StringComparer.OrdinalIgnoreCase);
-
-            return CreateWwwAuthenticateParameters(parameters);
+            return CreateFromWwwAuthenticationHeaderValue(wwwAuthenticateValue, string.Empty);
```

Comment:
Doesn't this obsolete API only handle Bearer? Why do you pass string.empty here?